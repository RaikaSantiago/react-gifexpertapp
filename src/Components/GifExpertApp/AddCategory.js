import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategoria }) => {
  
  const [first, setfirst] = useState('');

  const handInputChange = (e) => {
      return setfirst(e.target.value);
  }

  const handleSubmit = (e) => {
      e.preventDefault();
      if (first.trim().length > 0) {
        setCategoria(cat => [first,...cat ]);
        setfirst('');
      }
  }
  return (
    <form onSubmit={handleSubmit}>
        <h2>AddCategory</h2>
        <input  type="text" value={first} onChange={handInputChange} />
    </form>   
  )
}


AddCategory.propTypes = {
  setCategoria: PropTypes.func.isRequired
}

