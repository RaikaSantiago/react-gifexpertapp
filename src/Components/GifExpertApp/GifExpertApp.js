import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid/GifGrid';

export const GifExpertApp = () => {

  const categoriasArr = ['Homero'];
  const [categorias, setCategoria] = useState(categoriasArr);
  const handleAdd = () => {
    // setCategoria(['p4',...categorias]);
    setCategoria(cat => [ 'p4',...cat]);
  }

  return (
    <>  
        <h2>GifExpertApp</h2>
        <AddCategory  setCategoria={ setCategoria } />
        <hr />
        <button onClick={ handleAdd }>Agregar</button>
          { 
            categorias.map((cat, index) => (
              <GifGrid key={index} cat={cat} />
            ))
          }
    </>
  )
}


