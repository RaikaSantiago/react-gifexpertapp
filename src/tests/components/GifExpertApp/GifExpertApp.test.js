import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../../../Components/GifExpertApp/GifExpertApp';

describe('Pruebas <GifExpertApp  />', () => { 

    let wrapper = shallow( <GifExpertApp  />);
    beforeEach( () => {
        wrapper = shallow( <GifExpertApp  />);
    })

    test('Debe de mostrar <GifExpertApp  /> correctamente', () => { 

        expect( wrapper).toMatchSnapshot();
    })

})