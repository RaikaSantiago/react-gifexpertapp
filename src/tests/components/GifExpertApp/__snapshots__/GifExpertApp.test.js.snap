// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifExpertApp  /> Debe de mostrar <GifExpertApp  /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategoria={[Function]}
  />
  <hr />
  <button
    onClick={[Function]}
  >
    Agregar
  </button>
  <GifGrid
    cat="Homero"
    key="0"
  />
</Fragment>
`;
